portfolio/src/main/webapp/script.js

```diff
@@ -12,39 +12,70 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/** @param slideIndex slide currently being displayed */
 var slideIndex = 1;
+
+/**
+ * Supported class states for dots and slides.
+ * @enum {string}
+ */
+const states = {
+  SHOW_DOT: ' active',
+  HIDE_DOT: '',
+  SHOW_SLIDE: 'block',
+  HIDE_SLIDE: 'none',
+}
+
 showSlides(slideIndex);
 
-// Next and previous controls
+/**
+ * Function computes which slide should be shown and class showSlides on that
+ * specific slide. Function is called when either arrows are pressed.
+ * @param {integer} n where n = 1 to go forward or n = -1 to go back
+ */
 function plusSlides(n) {
```

Comment:
Optional: consider passing an enum where the only possible values are 1 and -1.

you could also add a TODO to do this in a separate change.

Reply:
Added a TODO in index.html to see if its possible to add an enum.